### PR TITLE
read body on resource uploads

### PR DESCRIPTION
**Pull request by jrwren**
apache httpd does not proxy responses if the request body is not
entirely read. This is already done differently in charm upload.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1910/
Test PASSed.

**Review by frankban — APPROVED** [review]
Very nice, thanks!

File: internal/v5/api.go [hunk 1]
```diff
@@ -1580,6 +1581,20 @@ func (h *ReqHandler) ResolvedIdHandler(f ResolvedIdHandler, cacheFields ...strin
...
+// reqBodyReadHandler returns an id handler that reads the request body
+// before returning a response.
+func reqBodyReadHandler(f router.IdHandler) router.IdHandler {
+	return func(id *charm.URL, w http.ResponseWriter, req *http.Request) error {
+		r := f(id, w, req)
+		count, err := io.Copy(ioutil.Discard, req.Body)
+		if err != nil {
+			logger.Errorf("error discarding request body %s", err)
+		}
+		logger.Debugf("discarded %d bytes from request Body", count)
```
Comment (frankban) [inline internal/v5/api.go]: s/Body/body/

**Review by bac — APPROVED** [review]

**Comment by jrwren** [comment]
:shipit:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1911/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**